src/adr/0043-avoid-elasticsearch-on-critical-reads.md

```diff
@@ -156,4 +156,6 @@ PRIMARY KEY mailboxId
 CLUSTERING COLUMN messageId
 COLUMN sentAt
 COLUMN receivedAt
-```
\ No newline at end of file
+```
+
+Note that to handle position & limit, we need to fetch `position + limit` ordered items then removing `position` firsts items.
```

Comment:
so if I scroll quickly n times, I will generate 1+2+...+n = n*(n+1)/2 cassandra requests ~= O(n²)

that's pretty bad, no? Couldn't it be a cause of ElasticSearch slowness? Could it slow down Cassandra?

Reply:
True for Cassandra.

True for ElasticSearch.

JMAP includes some limits concurent call, rate limiting - that can help mitigating these concenrs in the future.

>  Couldn't it be a cause of ElasticSearch slowness?

Maybe for some.

I succeeded to clearly link some to reindexing as well thanks to @tuanlc .